scripts/linters/python_linter.py

```diff
@@ -117,6 +117,44 @@ def _check_import_order(self):
                 summary_messages.append(summary_message)
         return summary_messages
 
+    def _check_non_test_files(self):
+        """This function is used to check that function
+           with test_only in their names are in test files.
+        """
+        if self.verbose_mode_enabled:
+            python_utils.PRINT('Starting function defintion checks')
+            python_utils.PRINT('----------------------------------------')
+        summary_messages = []
+        files_to_check = self.py_filepaths
+        with linter_utils.redirect_stdout(sys.stdout):
+            failed = False
+            for filepath in files_to_check:
+                if not filepath.endswith('_test.py'):
+                    file_content = FILE_CACHE.read(filepath)
+                    for line_number, line in enumerate(
+                            file_content.split('\n')):
+                        ind1 = line.find(' def ')
+                        ind2 = line.find('test_only')
+                        if ind1 != -1 and ind2 != -1:
+                            python_utils.PRINT(
+                                'ERROR:   Non test file uses test functions at'
+                                ' line:', line_number + 1, 'in file:', filepath)
+                            failed = True
+
+            if failed:
+                summary_message = (
+                    '%s   Function defintion checks failed,'
+                    'see affect files above.' % (_MESSAGE_TYPE_FAILED))
+                python_utils.PRINT(summary_message)
+                summary_messages.append(summary_message)
```

Comment:
Use these lines only once after the `else` block.

Reply:
done